#!/usr/bin/env python
PACKAGE = "reconfros_software"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_speed", double_t, 0, "Speed for linear movement (forward)", 0.7, 0.0, 5.0)
gen.add("angular_speed_factor", double_t, 0, "Speed Factor for angular movement (z-axe)", 0.025, 0.0, 20.0)
gen.add("angular_speed_max", double_t, 0, "Max. angular speed", 0.5, 0.0, 20.0)
gen.add("stop_step", double_t, 0, "Decreases speed with this value when the robot needs to stop", 0.3, 0.0, 5.0)
gen.add("drive_threshold", double_t, 0, "If correction value value is larger than this threshold the robot stops", 20.0, 0.0, 360.0)
gen.add("resume_threshold", double_t, 0, "If robot had to stop it resumes driving forward after reaching this threshold", 5.0, 0.0, 360.0)

exit(gen.generate(PACKAGE, "navigation", "NAV"))
